p256k1/src/scalar.rs

```diff
@@ -151,18 +150,17 @@ impl Scalar {
             return ret;
         }
 
-        let log = n.ilog2() + 1;
-        let mut i: u32 = 0;
+        let mut i: u32 = 1;
         for byte in n.to_le_bytes() {
             let bits = byte.view_bits::<Lsb0>();
             for bit in bits {
                 if *bit {
                     ret *= square;
                 }
                 if {
-                    i += 1;
+                    i <<= 1;
                     i
-                } > log
+                } > n
```

Comment:
BTW, the algorithm time depends on `n` (more leading zero, faster running), can it be used in timing attack https://en.wikipedia.org/wiki/Timing_attack ?

Reply:
The code was already non-constant time, so not relevant to this PR.  And the general `Scalar ^ Scalar` case doesn't have a short circuit, so also not relevant to that.

Timing attacks are real, so if we ever use this for an algorithm where we're doing `Scalar ^ u32` where the exponent is not constant, then yes there will be the potential for a timing attack there.  Something to keep in mind for that future.